spec/install/gemfile/path_spec.rb

```diff
@@ -484,6 +484,68 @@
 
       expect(the_bundle).to include_gems "rack 1.0.0"
     end
+
+    it "keeps using the same version if it's compatible" do
+      build_lib "foo", "1.0", :path => lib_path("foo") do |s|
+        s.add_dependency "rack", "0.9.1"
+      end
+
+      bundle "install"
+
+      expect(the_bundle).to include_gems "rack 0.9.1"
+
+      lockfile_should_be <<-G
+        PATH
+          remote: #{lib_path("foo")}
+          specs:
+            foo (1.0)
+              rack (= 0.9.1)
+
+        GEM
+          remote: #{URI.parse("file://#{gem_repo1}/")}
+          specs:
+            rack (0.9.1)
+
+        PLATFORMS
+          #{lockfile_platforms}
+
+        DEPENDENCIES
+          foo!
+
+        BUNDLED WITH
+           #{Bundler::VERSION}
+      G
+
+      build_lib "foo", "1.0", :path => lib_path("foo") do |s|
+        s.add_dependency "rack"
+      end
+
+      bundle "install"
+
+      lockfile_should_be <<-G
+        PATH
+          remote: #{lib_path("foo")}
+          specs:
+            foo (1.0)
+              rack
+
+        GEM
+          remote: #{URI.parse("file://#{gem_repo1}/")}
```

Comment:
It's off topic though, I've found `source` with file uri generates weird Gemfile.lock (`remote: file:/dummy/`) on Ruby 2.4

https://travis-ci.org/bundler/bundler/jobs/525810789

### Gemfile

```
source "file:///dummy"
```

### Gemfile.lock

```
GEM
  remote: file:/dummy/
  specs:

PLATFORMS
  ruby
  x86_64-linux

DEPENDENCIES

BUNDLED WITH
   2.1.0.pre.1
```

Reply:
I guess it's because of this difference:

```
$ docker run --rm ruby:2.4 ruby -e 'puts URI.parse("file:///dummy")'
file:/dummy
```

```
$ docker run --rm ruby:2.5 ruby -e 'puts URI.parse("file:///dummy")'
file:///dummy
```

That's why I added `URI.parse` to my spec to pass test on both Ruby <=2.4 and >=2.5

Reply:
Not sure if it's related but we have a helper doing some normalization, `lockfile_should_be`, but I'm noticing that it always receives `file://` sources as `file://localhost#{gem_repo1}/`.

Reply:
> I'm noticing that it always receives `file://` sources as `file://localhost#{gem_repo1}/`.

That's the point.
I want to share this context and avoid changing which affects another test to keep diff simple:

https://github.com/bundler/bundler/blob/f4c79f2744081b8a60942642786a79d7e7add6a2/spec/install/gemfile/path_spec.rb#L472-L475